compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3358,6 +3358,8 @@ object Types {
           absMems.find(_.symbol.name == nme.apply)
         else None
       }
+      else if (tp.isInstanceOf[HKApply])
+        unapply(tp.asInstanceOf[HKApply].superType)
       else None
```

Comment:
Thanks @odersky , indeed the code should go to `zeroParamClass`, fixed in the new commit.